apps/gk_neut_species.c

```diff
@@ -671,6 +702,35 @@ gk_neut_species_init(struct gkyl_gk *gk, struct gkyl_gyrokinetic_app *app, struc
     upper_vel[d] = s->info.upper[d];
     ghost_vel[d] = 0; // no ghost-cells in velocity space
   }
+
+  // Define phase basis
+  if (app->use_gpu) {
+    // allocate device basis if we are using GPUs
+    s->basis_on_dev = gkyl_cu_malloc(sizeof(struct gkyl_basis));
+  }
+  else {
+    s->basis_on_dev = &app->basis;
+  }
+
+  // Basis is tensor for p=1 and ser for p>1
+  if (app->poly_order > 1) {
+    gkyl_cart_modal_serendip(&s->basis, pdim, app->poly_order);
+  }
+  else if (app->poly_order == 1) {
+    gkyl_cart_modal_hybrid(&s->basis, cdim, vdim); // p=2 in v for neutral species
+    // gkyl_cart_modal_tensor(&s->basis, pdim, app->poly_order); // for canonical PB
+  }
+
+  if (app->use_gpu) {
+    if (app->poly_order > 1) {
+      gkyl_cart_modal_serendip_cu_dev(s->basis_on_dev, pdim, app->poly_order);
+    }
+    else if (app->poly_order == 1) {
+      gkyl_cart_modal_hybrid_cu_dev(s->basis_on_dev, cdim, vdim); // p=2 in v for neutral species
+      // gkyl_cart_modal_tensor_cu_dev(s->basis_on_dev, pdim, app->poly_order); // for canonical PB
+    }
+  }
+  
```

Comment:
Here you have basis_on_dev allocated and assigned in two separate if-statements. Why not just put it in one if-statement after you assign s->basis?

also, delete line 730 if not used